src/osd/OSDMap.cc

```diff
@@ -5562,7 +5562,13 @@ void OSDMap::check_health(health_check_map_t *checks) const
         if (pool.has_flag(pg_pool_t::FLAG_FULL_QUOTA)) {
           // may run out of space too,
           // but we want EQUOTA taking precedence
-          ss << "pool '" << pool_name << "' is full (running out of quota)";
+	  if (pool.has_flag(pg_pool_t::FLAG_FULL_QUOTA_OBJECTS)) {
+	    ss << "pool '" << pool_name << "' is full (reached objects quota "
+	       << pool.quota_max_objects << ")";
+	  } else {
```

Comment:
the user is losing the fact that we may also have reached full bytes quota though. Yes, either one will represent some form of quota being exhausted, but if I were to get just the objects one, I could easily assume that was the only thing I needed to look for (even if I also reached the bytes full quota).

Reply:
Yeah, I'd like to fix that as well but it'd be quite a bit more complicated.

Well, I guess we could add another flag POOL_QUOTA_FULL_BYTES and use the POOL_QUOTA_FULL as an indicator for either BYTES or OBJECTS. Would people prefer that?

Reply:
I think that would make sense. Either that, or adjust the message you're outputting to be more verbose? E.g., `(reached objects quota) || (reached bytes quota) || (reached objects and bytes quota)`. It would make the code a bit more annoying, but I think that could work out? Alternatively, use `FLAG_FULL_QUOTA` for bytes, and `FLAG_FULL_QUOTA_OBJECTS` for objects, but make sure to output them according to what's full, may it be one of them or both?

Reply:
I see two options:
- add POOL_QUOTA_FULL_BYTES.  safer, but spends more bits than we need (3 bits for only 4 states)
- do the original rename of POOL_QUOTA_FULL -> POOL_QUOTA_BYTES with the compat-fu.

I'm not too worried about the pool flags.  I'm also not too worried about the compat encoding issues, since we're pretty good at that for OSDMap now.  But if we are going ot backport this to nautilus, then let's go the safe route.. silly to spend a feature bit to save a pool flag bit.

Reply:
Okay, I've updated as discussed here.